### ISSUE TITLE
GPU tests fail out of memory

### ISSUE THREAD

**Issue by brianprichardson** [deepmind/open_spiel#494]
Build looks fine on Ubuntu 18.04 after changing script to use CXX=clang++-10 and commenting out the if test in build_and_run_tests.sh

Some GPU tests fail due to out of memory as several running at once with 
MAKE_NUM_PROCS=$(${NPROC})
let TEST_NUM_PROCS=4*${MAKE_NUM_PROCS}

Changed that to just
let TEST_NUM_PROCS=1
and re-ran the failed tests individually by changing
ArgsLibAddArg test_only string "python_deep_cfr_tf2_test" 'Builds and runs the specified test only (use "all" to run all tests)'
and they all passed.

Perhaps some GPU memory fencing would be helpful.
2080ti has 11GB VRAM.
The tests seem to individually max that out, so only one at a time can run.

**Comment by lanctot** [comment]
Hi @brianprichardson ,

Thanks. These tests were intended to run only on CPU. 

Can you tell me how you ran on GPU? I wonder if we can detect that dynamically and set the num_procs to 1 in that case. I would have expected you'd have to make several individual changes to the files to run on GPU.

**Comment by brianprichardson** [comment]
I'm not entirely sure what I did.
I already had an existing venv created that exact name for other things.
Some of the versions in that venv are
tensorboard-plugin-wit  1.7.0
termcolor               1.1.0
tf-estimator-nightly    2.4.0.dev2020102301
tf-nightly-gpu          2.5.0.dev20201024
torch                   1.5.0+cu101
torchsummary            1.5.1
torchvision             0.6.0+cu101

Maybe that gpu-only version was somehow used.

Attached is part of the install and test that I ran.

[BPR-open_spiel-log.txt](https://github.com/deepmind/open_spiel/files/5953645/BPR-open_spiel-log.txt)
